### DISCUSSION TITLE
Why do I fetch data server side and not client side?

### DISCUSSION THREAD

**Discussion by Niocas** [vikejs/vike#957] [Discussion] [answered]
In the documentation it says for SSR, I should use myPage.page.server.js to get my data from my api.
Is that so the server can render the page with all the data, so the data is inside the source code, if a crawler
would look at it? (Would that mean users which inspect my source code, can't see what api I am using, because
it is on the server side?)

If it would take like 2 seconds to load the data, would the page not "respond/load" for
2 seconds without giving any information about that the data is loading? 

And do pages get rerendered every single time a user visits my page? (So 1000 visits = 1000 renders)

**Comment by brillout** [comment, marked as answer]
I recommend to tinker with VPS and get your hands dirty  &mdash; you'll discover how it all works.
